opensearch/src/main/java/org/opensearch/sql/opensearch/data/type/OpenSearchDataType.java

```diff
@@ -230,6 +234,9 @@ public String legacyTypeName() {
     if (mappingType == null) {
       return exprCoreType.typeName();
     }
+    if (mappingType.toString().equalsIgnoreCase("DATE")) {
```

Comment:
change to mappingType == Date?

Reply:
Should we check mappingType == DateNanos either?
```java
if (mappingType == Date || mappingType == DateNanos) {
```